mesonbuild/modules/_qt.py

```diff
@@ -91,10 +95,101 @@ class CompileTranslationsKwArgs(TypedDict):
+        module_version: str
+        module_prefix: str
+        qml_sources: T.Sequence[T.Union[FileOrString, build.CustomTarget, build.CustomTargetIndex]]
+        qml_singletons: T.Sequence[T.Union[FileOrString, build.CustomTarget, build.CustomTargetIndex]]
+        qml_internals: T.Sequence[T.Union[FileOrString, build.CustomTarget, build.CustomTargetIndex]]
+        designer_supported: bool
+        imports: T.List[str]
+        optional_imports: T.List[str]
+        default_imports: T.List[str]
+        depends_imports: T.List[str]
+        output: str
+
+    class GenQmlCachegenKwArgs(TypedDict):
+
+        target_name: str
+        qml_sources: T.Sequence[T.Union[FileOrString, build.CustomTarget]]
+        qml_qrc: T.Union[FileOrString, build.CustomTarget]
+        extra_args: T.List[str]
+        module_prefix: str
+        method: str
+
+    class GenQmlTypeRegistrarKwArgs(TypedDict):
+
+        target_name: str
+        import_name: str
+        major_version: str
+        minor_version: str
+        namespace: str
+        typeinfo: str
+        generate_qmltype: bool
+        module_path_outdir: str
+        collected_json: T.Optional[T.Union[FileOrString, build.CustomTarget]]
+        extra_args: T.List[str]
+        method: str
+
+    class MocJsonCollectKwArgs(TypedDict):
+
+        target_name: str
+        moc_json: T.Sequence[build.GeneratedList]
+        method: str
+
+    class QmlModuleKwArgs(TypedDict):
+
+        qml_sources: T.Sequence[T.Union[File, str, build.CustomTarget]]
+        qml_singletons: T.Sequence[T.Union[File, str, build.CustomTarget]]
+        qml_internals: T.Sequence[T.Union[File, str, build.CustomTarget]]
+        resources_prefix: str
+        output_directory: str
+        moc_sources: T.Sequence[T.Union[FileOrString, build.CustomTarget, build.CustomTargetIndex, build.GeneratedList]]
+        moc_headers: T.Sequence[T.Union[FileOrString, build.CustomTarget, build.CustomTargetIndex, build.GeneratedList]]
+        include_directories: T.List[T.Union[str, build.IncludeDirs]]
+        imports: T.List[str]
+        optional_imports: T.List[str]
+        default_imports: T.List[str]
+        depends_imports: T.List[str]
+        designer_supported: bool
+        namespace: str
+        typeinfo: str
+        moc_extra_args: T.List[str]
+        qmlcachegen_extra_args: T.List[str]
+        qmltyperegistrar_extra_args: T.List[str]
+        generate_qmldir: bool
+        generate_qmltype: bool
+        create_plugin_target: bool
+        generate_plugin_source: bool
+        cachegen: bool
+        dependencies: T.List[T.Union[Dependency, ExternalLibrary]]
+        method: str
+        preserve_paths: bool
+
+def _list_in_set_validator(choices: T.Set[str]) -> T.Callable[[T.List[str]], T.Optional[str]]:
+    """Check that the choice given was one of the given set."""
+    def inner(checklist: T.List[str]) -> T.Optional[str]:
+        for check in checklist:
+            if check not in choices:
```

Comment:
I think you can simplify (and make a better error message) this to something like:
```python
invalid = set(checklist).difference(choices)
if invalid:
    return f"invalid selections {', '.join(sorted(invalid))}, valid elements are {', '.join(sorted(choices))}."